packages/repack/ios/ScriptManager.mm

```diff
@@ -257,12 +251,55 @@ - (void)executeFromFilesystem:(RCTCxxBridge *)bridge
             filesystemScriptUrl = [[NSBundle mainBundle] URLForResource:scriptName withExtension:scriptExtension];
         }
         NSData *data = [[NSData alloc] initWithContentsOfFile:[filesystemScriptUrl path]];
-        [bridge executeApplicationScript:data url:url async:YES];
-        resolve(nil);
+        [self evaluateJavascript:data url:filesystemScriptUrl resolve:resolve reject:reject];
     } @catch (NSError *error) {
-        reject(CodeExecutionFromFileSystemFailure, error.localizedDescription, nil);
+        reject(CodeExecutionFailure, error.localizedDescription, nil);
+    }
+}
+
+- (facebook::jsi::Runtime *)getJavaScriptRuntimePointer
+{
+    if (!self.bridge.runtime) {
+        return nil;
     }
     
+    facebook::jsi::Runtime *jsRuntime = (facebook::jsi::Runtime *)self.bridge.runtime;
+    return jsRuntime;
+}
+
+- (void)evaluateJavascript:(NSData *)code
+                       url:(NSURL *)url
+                   resolve:(RCTPromiseResolveBlock)resolve
+                    reject:(RCTPromiseRejectBlock)reject
+{
+    std::shared_ptr<facebook::react::CallInvoker> callInvoker = self.bridge.jsCallInvoker;
+    if (!callInvoker) {
+        reject(CallInvokerUnavailableError, @"Missing CallInvoker - bridgeless on RN 0.73 is not supported", nil);
+        return;
+    }
+    
+    // There is no backwards compatible way to get runtime in an asynchronous manner
+    // we obtain the ptr here, and use it inside of invokeAsync callback
+    // TODO: Consider doing this during initialization step once to fail fast
+    facebook::jsi::Runtime *runtime = [self getJavaScriptRuntimePointer];
+    if(!runtime) {
+        reject(RuntimeUnavailableError, @"Can't access React Native runtime", nil);
+        return;
+    }
+    
+    std::string source(static_cast<const char*>([code bytes]), [code length]);
+    std::string sourceUrl([[url absoluteString] UTF8String]);
```

Comment:
nit: I think more modern way to init stuff is with `{}` `std::string sourceUrl { [[url absoluteString] UTF8String] };`

Reply:
oh yeah sure! apparently this (list initializer) disables the implicit conversion so the compiler can catch more errors this way! nice find!